File: app/src/main/java/nl/ictrek/ictrix/adapters/ChatListAdapter.java
package nl.ictrek.ictrix.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import nl.ictrek.ictrix.Chat;
import nl.ictrek.ictrix.R;

/**
 * Created by Koen Bolhuis on 11-Apr-17.
 */

public class ChatListAdapter extends RecyclerView.Adapter<ChatListAdapter.ViewHolder> {
    private List<Chat> mChatList;

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView imageView;
        public TextView textViewTitle, textViewSummary, textViewTime;
        public ViewHolder(View view) {
            super(view);
            imageView = (ImageView)view.findViewById(R.id.imageView_chatImage);
            textViewTitle = (TextView)view.findViewById(R.id.textView_chatTitle);
            textViewSummary = (TextView)view.findViewById(R.id.textView_chatSummary);
            textViewTime = (TextView)view.findViewById(R.id.textView_chatTime);
        }
    }

    public ChatListAdapter(List<Chat> chatList) {
        mChatList = chatList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.chat_list_chat, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Chat chat = mChatList.get(position);
        // TODO: handle image
        String title = chat.getTitle();
        if (chat.getType() == Chat.Type.GROUP) {
            holder.textViewTitle.setCompoundDrawablesWithIntrinsicBounds(
                    R.drawable.ic_people_black_24dp, 0, 0, 0
            );
            title = " " + title;
        }
        holder.textViewTitle.setText(title);
        holder.textViewSummary.setText(chat.getSummary());
        holder.textViewTime.setText(chat.getTime());
    }

    @Override
    public int getItemCount() {
        return mChatList.size();
    }
}

File: app/src/main/java/nl/ictrek/ananas/adapters/ChatListAdapter.kt
package nl.ictrek.ananas.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import kotlinx.android.synthetic.main.chat_list_chat.view.*

import nl.ictrek.ananas.Chat
import nl.ictrek.ananas.R

/**
 * Created by Koen Bolhuis on 11-Apr-17.
 *
 * listadapter for the chats on the main activity.
 *
 * @param mChatList List with the chats.
 */
class ChatListAdapter(private val mChatList: List<Chat>) : RecyclerView.Adapter<ChatListAdapter.ViewHolder>() {

    class ViewHolder(val view: View) : RecyclerView.ViewHolder(view)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder = ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.chat_list_chat, parent, false)
    )

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val chat = mChatList[position]
        // TODO: handle image
        var title = chat.title
        if (chat.type === Chat.Type.GROUP) {
            holder.view.textView_chatTitle.setCompoundDrawablesWithIntrinsicBounds(
                    R.drawable.ic_people_black_18dp, 0, 0, 0
            )
            title = " $title"
        }
        holder.view.textView_chatTitle.text = title
        holder.view.textView_chatSummary.text = chat.summary
        holder.view.textView_chatTime.text = chat.time
    }

    override fun getItemCount(): Int = mChatList.size
}
